crates/core/component/dex/src/component/candlestick/candlestick.rs

```diff
@@ -0,0 +1,56 @@
+use anyhow::Result;
+use serde::{Deserialize, Serialize};
+
+use penumbra_proto::{core::component::dex::v1 as pb, DomainType};
+
+#[derive(Debug, Clone, Copy, PartialEq, Serialize, Deserialize)]
+#[serde(try_from = "pb::CandlestickData", into = "pb::CandlestickData")]
+pub(crate) struct CandlestickData {
+    /// The height of the candlestick data.
+    pub(crate) height: u64,
+    /// The first observed price during the block execution.
+    pub(crate) open: f64,
+    /// The last observed price during the block execution.
+    pub(crate) close: f64,
+    /// The highest observed price during the block execution.
+    pub(crate) high: f64,
+    /// The lowest observed price during the block execution.
+    pub(crate) low: f64,
+    /// The volume that traded "directly", during individual position executions.
+    pub(crate) direct_volume: f64,
+    /// The volume that traded as part of swaps, which could have traversed multiple routes.
+    pub(crate) swap_volume: f64,
+}
```

Comment:
Does these need to be `pub(crate)`?

Reply:
Yes, otherwise the Chandelier can't instantiate one (unless we make a `new` method but it seems unneccessary)

Reply:
Sorry, what I meant was, why not make them just `pub`? There's no invariants we're maintaining right?